private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -0,0 +1,423 @@
+            JavaScopes.RUNTIME);
+
+    Set<Exclusion> excluded =
+        artifact.getExclusions().stream().map(this::createExclusion).collect(Collectors.toSet());
+
+    return bom.setScope("import").setExclusions(excluded);
+  }
+
+  private Dependency createDependency(
+      com.github.bazelbuild.rules_jvm_external.resolver.Artifact source) {
+    Artifact artifact;
+    Coordinates coords = source.getCoordinates();
+    artifact =
+        new DefaultArtifact(
+            coords.getGroupId(),
+            coords.getArtifactId(),
+            coords.getClassifier(),
+            coords.getExtension(),
+            coords.getVersion());
+
+    Set<Exclusion> excluded =
+        source.getExclusions().stream().map(this::createExclusion).collect(Collectors.toSet());
+
+    return new Dependency(artifact, JavaScopes.RUNTIME).setExclusions(excluded);
+  }
+
+  private Exclusion createExclusion(Coordinates coordinates) {
+    return new Exclusion(coordinates.getGroupId(), coordinates.getArtifactId(), "*", "*");
+  }
+
+  @Override
+  public Graph<Coordinates> resolve(ResolutionRequest request) {
+    List<RemoteRepository> repos =
+        request.getRepositories().stream()
+            .map(remoteRepositoryFactory::createFor)
+            .collect(Collectors.toList());
+
+    List<Dependency> boms =
+        request.getBoms().stream().map(this::createBom).collect(Collectors.toList());
+    List<Dependency> dependencies =
+        request.getDependencies().stream().map(this::createDependency).collect(Collectors.toList());
+    Set<Exclusion> globalExclusions =
+        request.getGlobalExclusions().stream()
+            .map(this::createExclusion)
+            .collect(Collectors.toSet());
+
+    RepositorySystem system = createRepositorySystem();
+    ConsoleRepositoryListener consoleLog = new ConsoleRepositoryListener(listener);
+    ErrorReportingListener errors = new ErrorReportingListener();
+    CoordinateGatheringListener coordsListener = new CoordinateGatheringListener();
+    RepositorySystemSession session =
+        prepareSession(
+            system,
+            new DefaultDependencyManager(),
+            new CompoundListener(consoleLog, errors, coordsListener),
+            request.getLocalCache());
+
+    List<RemoteRepository> repositories = new ArrayList<>(repos.size());
+    repositories.add(createRemoteRepoFromLocalM2Cache(request.getLocalCache()));
+    repositories.addAll(repos);
+
+    List<Dependency> amendedDeps = addGlobalExclusions(globalExclusions, dependencies);
+    List<Dependency> amendedBoms = addGlobalExclusions(globalExclusions, boms);
+
+    consoleLog.setPhase("Resolving BOM artifacts");
+
+    List<Dependency> managedDependencies =
+        resolveArtifactsFromBoms(system, session, repositories, amendedBoms);
+
+    managedDependencies = overrideDependenciesWithUserChoices(managedDependencies, dependencies);
+
+    Artifact fakeRoot = new DefaultArtifact("com.example:bazel-dep-resolver:1.0.0");
```

Comment:
Mind adding a comment what the fakeRoot is for?  I assume maven just requires something that look likes a root when it is resolving?

Reply:
Done